### PR TITLE
Initialize a new advanced metadata field for tags

### PR DESCRIPTION
**Pull request by ashmaroli**
Inspired by #380 

  - [x] create and delete tags
      - [x] create new tags by pressing `comma` key <kbd> , </kbd>
      - [x] delete a tag by clicking on the `x`
  - [x] prevent creating duplicate tags
  - [x] also create tags with <kbd> spacebar </kbd> or <kbd> Enter </kbd>
  - [x] delete the last tag with <kbd> Backspace </kbd>
  - [x] Update metadata spec suite
  - [x] Add `autocomplete` suggestions
      - [x] Update API to enable requesting `site.tags`
      - [x] update frontend to render a list of suggestions for new tags

### CODE REVIEW

**Comment by gsantner** [comment]
@ashmaroli Is this still a WIP or will this get discarded?

**Comment by ashmaroli** [comment]
@gsantner its still in WIP.. waiting for the API to mature enough to easily facilitate `get` on the array of `site.tags`.
This won't be discarded. Something will eventually come out of the hat.. :smiley:

File: src/components/metadata/MetaObject.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ export class MetaObject extends Component {
       let type = 'simple';
       if (_.isObject(value)) type = 'object';
       if (_.isArray(value)) type = 'array';
+      if (_.isArray(value) && key === 'tags') type = 'simple';
```
Comment (mertkahyaoglu) [inline src/components/metadata/MetaObject.js]: Could you extract these conditions into a method which returns type given value?

File: src/components/metadata/MetaSimple.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ import moment from 'moment';
 import momentLocalizer from 'react-widgets-moment';
 import FilePicker from '../FilePicker';
 import 'react-widgets/dist/css/react-widgets.css';
+import MetaTags from './MetaTags';
```
Comment (mertkahyaoglu) [inline src/components/metadata/MetaSimple.js]: Let's move this one line up.

**Comment by ashmaroli** [comment]
This feature lacks one thing currently: there's no way to select *existing tag*.. 
So was thinking of importing the `MultiSelect` component from `react-widgets`.. :man_shrugging:

**Comment by mertkahyaoglu** [comment]
That could be done in another pr maybe another release. This is quite enough IMO.

File: src/components/metadata/MetaObject.js (hunk 1 above)
Comment (ashmaroli) [reply src/components/metadata/MetaObject.js]: Good idea. Will handle this in a separate PR..

**Pull request merged**